sklearn/neighbors/classification.py

```diff
@@ -326,6 +332,24 @@ def __init__(self, radius=1.0, weights='uniform',
         self.weights = _check_weights(weights)
         self.outlier_label = outlier_label
 
+
+    def fit(self, X, y):
+        """Fit the model using X as training data and y as target values
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix, BallTree, KDTree}
+            Training data. If array or matrix, shape [n_samples, n_features],
+            or [n_samples, n_samples] if metric='precomputed'.
+
+        y : {array-like, sparse matrix}
+            Target values of shape = [n_samples] or [n_samples, n_outputs]
+
+        """
+        self.outlier_label = _check_outlier_label(self.outlier_label)
```

Comment:
You can't modify an input parameter according to our conventions, or else calling fit again with different data would do unexpected things. Instead set outlier_label_